mana_decay_provider.go

```diff
@@ -201,10 +201,10 @@ func (p *ManaDecayProvider) ManaGenerationWithDecay(amount BaseToken, slotIndexC
 }
 
 // RewardsWithDecay applies the decay to the given stored mana.
-func (p *ManaDecayProvider) RewardsWithDecay(rewards Mana, epochIndexReward EpochIndex, epochIndexClaimed EpochIndex) (Mana, error) {
-	if epochIndexReward > epochIndexClaimed {
-		return 0, ierrors.Wrapf(ErrWrongEpochIndex, "the reward epoch index was bigger than the claiming epoch index: %d > %d", epochIndexReward, epochIndexClaimed)
+func (p *ManaDecayProvider) RewardsWithDecay(rewards Mana, epochReward EpochIndex, epochClaimed EpochIndex) (Mana, error) {
```

Comment:
rename `epochReward` and `epochClaimed` to `rewardEpoch`and `claimedEpoch` to be in line with the other names where we always postfix with 'Epoch' or 'Slot'.

epochReward also sounds like the 'rewards for an epoch' rather than 'the epoch of a reward'